pkg/okta/authentication_middleware.go

```diff
@@ -78,13 +88,26 @@ func (f oktaMiddlewareFactory) newPrincipal(jwt *jwtverifier.Jwt) (*authenticati
 	jc508User := jwtGroupsContainsJobCode(jwt, f.code508User)
 	jcTRBAdmin := jwtGroupsContainsJobCode(jwt, f.codeTRBAdmin)
 
+	userAccount, err := userhelpers.GetOrCreateUserAccount(
+		ctx,
+		f.Store,
+		f.Store,
+		euaID,
+		true,
+		userhelpers.GetOktaAccountInfoWrapperFunction(userhelpers.GetOktaAccountInfo),
+	) //TODO, do we need to do anything with the user? Should we pass the id around?
```

Comment:
Is this `TODO` still needed? we're passing a reference of the user into the `EUAPrincipal` object, which seems like what this was maybe getting at?

Reply:
thanks I agree.